sys/include/net/gcoap.h

```diff
@@ -1126,6 +1126,23 @@ ssize_t gcoap_encode_link(const coap_resource_t *resource, char *buf,
 sock_dtls_t *gcoap_get_sock_dtls(void);
 #endif
 
+/**
+ * @brief   Get the header of a request from a @ref gcoap_request_memo_t
+ *
+ * @param[in] memo  A request memo. Must not be NULL.
+ *
+ * @return  The request header for the given request memo.
+ */
+static inline coap_hdr_t *gcoap_request_memo_get_hdr(const gcoap_request_memo_t *memo)
+{
```

Comment:
Not critical: what's your opinion on adding an `assert(memo != NULL)`?

Reply:
Would require to include `assert.h` and also, since the macro uses `RIOT_FILE_RELATIVE` the provided line would be wrong (it would point to the including C-file). That's why I intentially omitted it.

Reply:
On the premise that the compiler *really* inlines this inline function ..

Reply:
I think it will still take the line and file from code.